sdk/core/azure-core/azure/core/credentials.py

```diff
@@ -41,34 +46,37 @@ def get_token(
 AzureNamedKey = namedtuple("AzureNamedKey", ["name", "key"])
 
 
-__all__ = ["AzureKeyCredential", "AzureSasCredential", "AccessToken", "AzureNamedKeyCredential", "TokenCredential"]
+__all__ = [
+    "AzureKeyCredential",
+    "AzureSasCredential",
+    "AccessToken",
+    "AzureNamedKeyCredential",
+    "TokenCredential",
+]
 
 
-class AzureKeyCredential(object):
+class AzureKeyCredential:
     """Credential type used for authenticating to an Azure service.
     It provides the ability to update the key without creating a new client.
 
     :param str key: The key used to authenticate to an Azure service
     :raises: TypeError
     """
 
-    def __init__(self, key):
-        # type: (str) -> None
+    def __init__(self, key: str) -> None:
         if not isinstance(key, str):
             raise TypeError("key must be a string.")
-        self._key = key  # type: str
+        self._key: str = key
```

Comment:
`key` is known to be `str` - so this annotation is superfluous?

Reply:
Maybe, but does it hurt?

Reply:
Only to the extent that people like might wonder why it is there and if there is something going on behind the scenes that makes it a requirement to be explicit here :).

Reply:
Fine, I'll remove it :p